### ISSUE TITLE
False negative for `no-else-return` after `except`

### ISSUE THREAD

**Issue by nickdrozd** [PyCQA/pylint#7788]
### Bug description

This snippet returns in the `except` block, and therefore the `else` is unnecessary:

```python
def f() -> bool:
    try:
        print('asdf')
    except:  # pylint: disable = bare-except
        return False
    else:
        return True
```

It should raise `no-else-return`.

### Command used

```shell
`pylint asdf.py`
```

### Pylint output

```shell
No warnings raised
```

### Expected behavior

Should raise `no-else-return`

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.13.0-dev0
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
```

**Comment by nickdrozd** [comment]
A similar case with `raise`:

```python
    def _load_reporter_by_name(self, reporter_name: str) -> reporters.BaseReporter:
        name = reporter_name.lower()
        if name in self._reporters:
            return self._reporters[name]()

        try:
            reporter_class = _load_reporter_by_class(reporter_name)
        except (ImportError, AttributeError, AssertionError) as e:
            raise exceptions.InvalidReporterError(name) from e
        else:
            return reporter_class()
```

The `except` raises an exception, so the `else` is not needed.

**Comment by marcinnformula** [comment]
What?!

**Comment by Pierre-Sassoulas** [comment]
Could you open a new issue @marcinnformula, a comment on a closed issue is going to be forgotten. I don't think we should revert but we could introduce an option or a way to distinguish the two use cases.
